version.go

```diff
@@ -0,0 +1,34 @@
+package hcsshim
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+type SupportedSchemaVersionsStruct struct {
+	SupportedSchemaVersions []hcsschema.Version `json:"SupportedSchemaVersions,omitempty"`

```

Comment:
not sure if there is a better way to do this. am trying using map...

Reply:
here's another way to do it using map. i think this is better. please let me know your preference.

func GetHcsSchemaVersion() (schemaVersion *hcsschema.Version, err error) {
	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
	defer cancel()

	serviceProperties, err := hcs.GetServiceProperties(ctx, hcsschema.PropertyQuery{PropertyTypes: []string{"Basic"}})
	if err != nil {
		return nil, fmt.Errorf("failed to retrieve HCS schema version: %s", err)
	}

	fmt.Println("serviceProperties.Properties[0] = ", serviceProperties.Properties[0])

	var result map[string]interface{}
	if err := json.Unmarshal(serviceProperties.Properties[0], &result); err != nil {
		return nil, fmt.Errorf("failed to unmarshal HCS Schema Version: %s", err)
	}

	supportedSchemaVersions := result["SupportedSchemaVersions"]
	tmp := supportedSchemaVersions.([]interface{})[0]

	schemaVersion = &hcsschema.Version{}
	schemaVersion.Major = int32(tmp.(map[string]interface{})["Major"].(float64))
	schemaVersion.Minor = int32(tmp.(map[string]interface{})["Minor"].(float64))

	return
}

Reply:
this is part of schema, use array is fine.  And this definition should located in schema files, not here

Reply:
fix